dbt/include/global_project/macros/materializations/snapshots/strategies.sql

```diff
@@ -70,7 +70,16 @@
         ({{ snapshotted_rel }}.{{ columns.dbt_valid_from }} < {{ current_rel }}.{{ updated_at }})
     {%- endset %}
 
-    {% set scd_id_expr = snapshot_hash_arguments([primary_key, updated_at]) %}
+    {% if primary_key is iterable and primary_key is not string and primary_key is not mapping %}
+        {% set scd_args = [] %}
+        {% for key in primary_key %}
+            {{ scd_args.append(key) }}
+        {% endfor %}
+        {{ scd_args.append(updated_at) }}
+        {% set scd_id_expr = snapshot_hash_arguments(scd_args) %}
+    {% else %}
+        {% set scd_id_expr = snapshot_hash_arguments([primary_key, updated_at]) %}
```

Comment:
I would only set the variable parts of `scd_args` in the if block and then move the call to `snapshot_hash_arguments` and the inclusion of `updated_at` outside of the if block:

```sql
    {% set scd_args = [updated_at] %}
    {% if primary_key is iterable and primary_key is not string and primary_key is not mapping %}
        {% for key in primary_key %}
            {{ scd_args.append(key) }}
        {% endfor %}
    {% else %}
        {{ scd_args.append(primary_key) }}
    {% endif %}
    {% set scd_id_expr = snapshot_hash_arguments(scd_args) %}
```

Or even better, I would make this a python method since there is no templating:

```sql
    {% set scd_args = api.Relation.scd_args(primary_key, updated_at) %}
    {% set scd_id_expr = snapshot_hash_arguments(scd_args) %}
```

Reply:
Much better. Done.